File: B17 Math/171 Excel Sheet column Number/Solution.java
/*
Time Complexity:
Space Complexity:
https://leetcode.com/problems/excel-sheet-column-number/discuss/52154/Concise-java-solution-with-explanation.

"B" = 2
"BC" = (2)26 + 3
"BCM" = (2(26) + 3)26 + 13
*/ 

public class Solution {
    public int titleToNumber(String s) {
        if (s == null) return -1;
        int sum = 0;
        // for each loop so we don't need to mess with index values.
        for (char c : s.toUpperCase().toCharArray()) {
            sum *= 26;
            sum += c - 'A' + 1;
        }
        return sum;
    }
}

File: B17 Math/171 Excel Sheet column Number/Solution.c
/*
Time Complexity:O(n)
Space Complexity:
https://leetcode.com/problems/excel-sheet-column-number/discuss/1790101/C%2B%2B-oror-4-Lines-oror-Easy-To-UnderStand-oror-0ms-oror-100

"B" = 2
"BC" = (2)26 + 3
"BCM" = (2(26) + 3)26 + 13
*/ C++


class Solution {
public:
    int titleToNumber(string columnTitle) {
        int result = 0;
        for(char c : columnTitle)
        {
			//d = s[i](char) - 'A' + 1 (we used  s[i] -  'A' to convert the letter to a number like it's going to be C)

            int d = c - 'A' + 1;
            result = result * 26 + d;
        }
        return result;
    }
};